### PR TITLE
MM-57261: Use a valid database engine version for MySQL

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
According to the `describe-db-engine-versions` subcommand of the aws cli (the source of truth for the valid versions), the only patch version for 3.05 is 3.05.2, ~~but we're upgrading to the newer 3.06 nonetheless~~:
```
> aws --profile=mm-loadtest rds describe-db-engine-versions --engine aurora-mysql | jq '.DBEngineVersions[].EngineVersion'
"5.7.mysql_aurora.2.07.9"
"5.7.mysql_aurora.2.07.10"
"5.7.mysql_aurora.2.11.1"
"5.7.mysql_aurora.2.11.2"
"5.7.mysql_aurora.2.11.3"
"5.7.mysql_aurora.2.11.4"
"5.7.mysql_aurora.2.11.4"
"5.7.mysql_aurora.2.12.0"
"5.7.mysql_aurora.2.12.1"
"8.0.mysql_aurora.3.01.0"
"8.0.mysql_aurora.3.01.1"
"8.0.mysql_aurora.3.02.0"
"8.0.mysql_aurora.3.02.1"
"8.0.mysql_aurora.3.02.2"
"8.0.mysql_aurora.3.02.3"
"8.0.mysql_aurora.3.03.0"
"8.0.mysql_aurora.3.03.1"
"8.0.mysql_aurora.3.03.2"
"8.0.mysql_aurora.3.03.3"
"8.0.mysql_aurora.3.04.0"
"8.0.mysql_aurora.3.04.1"
"8.0.mysql_aurora.3.04.2"
"8.0.mysql_aurora.3.05.2"
"8.0.mysql_aurora.3.06.0"
```

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57261

### CODE REVIEW

File: deployment/terraform/assets/variables.tf [hunk 1]
```diff
@@ -54,7 +54,7 @@ variable "db_cluster_identifier" {
 variable "db_engine_version" {
   type = map(any)
   default = {
-    "aurora-mysql"      = "8.0.mysql_aurora.3.05.0"
+    "aurora-mysql"      = "8.0.mysql_aurora.3.06.0"
```
Comment (streamer45) [inline deployment/terraform/assets/variables.tf:57]: Can we use `3.05.2` instead? The idea was to use the same MySQL patch version as our min supported one which is `8.0.32`.

Comment (agarciamontoro) [reply deployment/terraform/assets/variables.tf:57]: Ah, of course, hadn't thought of that. Let me change it.

**Review by streamer45 — APPROVED** [review]
Thanks!

**Review by cpoile — APPROVED** [review]

**Comment by agarciamontoro** [comment]
I'll test this, merge and cherry-pick to `release-1.15`

**Comment by agarciamontoro** [comment]
DB correctly created with 3.05.2 version:
image

**Pull request merged**